packages/graphql/src/registry.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  GraphQLBoolean,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  GraphQLObjectType,
+  GraphQLScalarType,
+  GraphQLUnionType,
+  type GraphQLSchemaConfig,
+} from "graphql";
+
+export class GraphQLTypeRegistry {
+  // A map of type names to their corresponding GraphQL types
+  private types: Map<
+    string,
+    | GraphQLScalarType
+    | GraphQLObjectType
+    | GraphQLInputObjectType
+    | GraphQLEnumType
+    | GraphQLUnionType
+  > = new Map();
+
+  constructor() {}
+
+  /**
+   * Registers a new GraphQL type in the registry.
+   * @param name The name of the type as defined in the TypeSpec program.
+   * @param type The GraphQL type to register.
+   */
+  registerType(
```

Comment:
what would hold the intermediate state where we gather data in enter model and actually materialize the data in exitModel i.e. if I have something like:

```
model Cat {
    bestFriend: Cat;
}
```

I need to know that `Cat` has already been seen and when I enter `bestFriend`, I should pull the partially materialized `Cat` model and add `bestFriend` to it.